pkg/sqlite/query.go

```diff
@@ -571,75 +613,153 @@ func (s *SQLQuerier) GetImagesForBundle(ctx context.Context, csvName string) ([]
+	providedRows, err := s.db.QueryContext(ctx, providedQuery, registry.GVKType, entryID)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	provided = []*api.GroupVersionKind{}
 	for providedRows.Next() {
-		var groupName sql.NullString
-		var versionName sql.NullString
-		var kindName sql.NullString
-		var pluralName sql.NullString
+		var value sql.NullString
 
-		if err := providedRows.Scan(&groupName, &versionName, &kindName, &pluralName); err != nil {
+		if err := providedRows.Scan(&value); err != nil {
 			return nil, nil, err
 		}
-		if !groupName.Valid || !versionName.Valid || !kindName.Valid || !pluralName.Valid {
-			return nil, nil, err
+
+		if !value.Valid {
+			continue
 		}
+		prop := registry.GVKProperty{}
+		if err := json.Unmarshal([]byte(value.String), &prop); err != nil {
+			continue
+		}
+
 		provided = append(provided, &api.GroupVersionKind{
-			Group:   groupName.String,
-			Version: versionName.String,
-			Kind:    kindName.String,
-			Plural:  pluralName.String,
+			Group:   prop.Group,
+			Version: prop.Version,
+			Kind:    prop.Kind,
 		})
+		groups[prop.Group] = struct{}{}
+		versions[prop.Version] = struct{}{}
+		kinds[prop.Kind] = struct{}{}
 	}
 	if err := providedRows.Close(); err != nil {
 		return nil, nil, err
 	}
 
-	requiredQuery := `SELECT DISTINCT api.group_name, api.version, api.kind, api.plural FROM api
-		 	  		  INNER JOIN channel_entry ON channel_entry.operatorbundle_name = api_requirer.operatorbundle_name
-		 	  		  INNER JOIN operatorbundle ON operatorbundle.name=channel_entry.operatorbundle_name
-		 	  		  INNER JOIN api_requirer ON (api.group_name=api_requirer.group_name AND api.version=api_requirer.version AND api.kind=api_requirer.kind AND operatorbundle.name=api_requirer.operatorbundle_name)
-			  		  WHERE channel_entry.entry_id=?`
+	requiredQuery := `SELECT DISTINCT dependencies.value FROM dependencies
+					  INNER JOIN channel_entry ON channel_entry.operatorbundle_name = dependencies.operatorbundle_name
+					  WHERE dependencies.type=? AND channel_entry.entry_id=?`
 
-	requiredRows, err := s.db.QueryContext(ctx, requiredQuery, entryID)
+	requiredRows, err := s.db.QueryContext(ctx, requiredQuery, registry.GVKType, entryID)
 	if err != nil {
 		return nil, nil, err
 	}
 	required = []*api.GroupVersionKind{}
 	for requiredRows.Next() {
+		var value sql.NullString
+
+		if err := requiredRows.Scan(&value); err != nil {
+			return nil, nil, err
+		}
+		if !value.Valid {
+			continue
+		}
+		dep := registry.GVKDependency{}
+		if err := json.Unmarshal([]byte(value.String), &dep); err != nil {
+			continue
+		}
+
+		required = append(required, &api.GroupVersionKind{
+			Group:   dep.Group,
+			Version: dep.Version,
+			Kind:    dep.Kind,
+		})
+		groups[dep.Group] = struct{}{}
+		versions[dep.Version] = struct{}{}
+		kinds[dep.Kind] = struct{}{}
+	}
+	if err := requiredRows.Close(); err != nil {
+		return nil, nil, err
+	}
+
+	argsFor := func(s map[string]struct{}) string {
```

Comment:
this is a lot of extra work to grab the plurals from the api table - would like to argue that we don't need to bother with it, but for now I just made it completely backwards-compatible just in case.